src/datasets/arrow_dataset.py

```diff
@@ -323,6 +318,22 @@ def from_file(
             indices_table=indices_pa_table,
         )
 
+    @classmethod
+    def from_parquet(cls, filename: str, in_memory: bool = True):
+        """Create Dataset from Parquet file.
+
+        Args:
+            filename (:obj:`str`): Path of the Parquet file.
+            in_memory (:obj:`bool`, default ``True``): Whether to copy the data in-memory.
+
+        Returns:
+            :class:`Dataset`
+        """
+        from .arrow_reader import ParquetReader
```

Comment:
The ParquetReader unfortunately always bring data in memory. The memory-map option is here only for reading performance. Under the hood the data from parquet are compressed so you can't just memory map the file since there's no on-the-fly decoding afaik.

I think the best option would be to implement a ArrowBasedBuilder as we did for csv/json/etc.